### ISSUE TITLE
Avoid use of Locale constructor in default locale extension

### ISSUE THREAD

**Issue by beatngu13** [junit-pioneer/junit-pioneer#658]
The `Locale` constructors have been [deprecated](https://download.java.net/java/early_access/loom/docs/api/java.base/java/util/Locale.html#%3Cinit%3E(java.lang.String)) in Java 19. From the constructor implementation notes:

>* Obsolete ISO 639 codes ("iw", "ji", and "in") are mapped to their current forms. See [Legacy language codes](https://download.java.net/java/early_access/loom/docs/api/java.base/java/util/Locale.html#legacy_language_codes) for more information.
>* For backward compatibility reasons, this constructor does not make any syntactic checks on the input.

Several [alternatives](https://download.java.net/java/early_access/loom/docs/api/java.base/java/util/Locale.html#ObtainingLocale) exist to obtain a `Locale`, but, e.g., [`Locale.Builder`](https://download.java.net/java/early_access/loom/docs/api/java.base/java/util/Locale.Builder.html) does syntax checks, which leads to the following test failures:

```text
DefaultLocaleTests
  applied on the method level
    sets the default locale using a language, a country and a variant
      java.util.IllformedLocaleException: Ill-formed variant: gb [at index 0]
    sets the default locale using a language
      java.util.IllformedLocaleException: Ill-formed language: en_EN [at index 0]
```

We should avoid the use of the deprecated `Locale` constructors and switch to one of the aforementioned alternatives. Note that this is a _breaking change_, as we change the behavior of the default locale extension.

**Comment by Bukama** [comment]
Locale is the same for 8 and 11 with
* Constructors (with language/country/variant)
* Factory-method "forLanguage(language-String), which takes IETF BCP 47 language strings, but does not makes syntax checks on the strings (as far as I understand the docs)
* Constants, e.g. Locale.US
* Builder which allows setting language/country/variant but with checking the " syntax requirements defined by the Locale class"

In 19 (didn't check version between) there's also a factory method "of" for language/country/variant. But the constructors are removed (as @beatngu13  already mentioned).

So question is now: which way do we use?
I personally like constants most as I don't have to care about the string values. But I think for our test cases the builder way makes more sense - meaning we have to change some cases to match valid IETF BCP 47 strings.

**Comment by beatngu13** [comment]
Closed via #662.
